Force sync inventory after PrepareResultEvent

Sytm opened the pull request:
This is required because when changing the result for an anvil (when the event is called because of AnvilMenu.setRenameText method) to an invalid result the item is not synced back to the client

Malfrador reviewed (approved):
Looks good to me and works as expected. Considering the client will refuse to pick-up results that don't match the input, forcing a re-sync of the result is necessary to make the event work properly in other cases too.

Commit:
Force sync inventory after PrepareResultEvent

This is required because when changing the result for an anvil (when the event is called because of setRenameText method) to an invalid result the item is not synced back to the client
